oauth2/oauth2-resource-server/src/main/java/org/springframework/security/oauth2/server/resource/authentication/JwtAuthenticationConverter.java

```diff
@@ -65,4 +73,14 @@ public void setJwtGrantedAuthoritiesConverter(Converter<Jwt, Collection<GrantedA
 		Assert.notNull(jwtGrantedAuthoritiesConverter, "jwtGrantedAuthoritiesConverter cannot be null");
 		this.jwtGrantedAuthoritiesConverter = jwtGrantedAuthoritiesConverter;
 	}
+
+	/**
+	 * Sets the principal claim name.
+	 *
+	 * @param principalClaimName The principal claim name
+	 * @since 5.2
+	 */
+	public void setPrincipalClaimName(String principalClaimName) {
+		this.principalClaimName = principalClaimName;
```

Comment:
Would you please add an `Assert` here to validate that the claim `hasText`?

Reply:
Also, would you please add a test to `JwtAuthenticationConverterTests` that confirms that the input validation works?

Reply:
Added validation and tests.